src/main/java/org/isf/menu/rest/UserController.java

```diff
@@ -358,5 +369,175 @@ public ResponseEntity<List<LitePermissionDTO>> retrievePermissionsByUsername(@Pa
+	 * @throws OHServiceException.
+	 */
+	@GetMapping(value = "/users/settings/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<UserSettingDTO> getUserSettingById(@PathVariable(name = "id")int id) throws OHServiceException {
+		LOGGER.info("Retrieve the userSetting By id {}:", id);
+		Optional<UserSetting> userSetting = userSettingManager.getUserSettingById(id);
+		if (!userSetting.isPresent()) {
+		    LOGGER.info("No user settings with id {}", id);
+	            return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
+		}
+		return ResponseEntity.ok(userSettingMapper.map2DTO(userSetting.get()));
+	}
+	
+	/**
+	 * Retrieves an existing {@link UserSettingDTO} by user.
+	 * 
+	 * @param userName - the name of user.
+	 * @param configName - the name of the userSetting {@link UserSetting} . 
+	 * @return {@link UserSettingDTO} if the UserSetting exist , null otherwise.
+	 * @throws OHServiceException.
+	 */
+	@GetMapping(value = "/users/{userName}/settings/{configName}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<UserSettingDTO> getUserSettingByUser(@PathVariable(name = "userName") String userName, @PathVariable(name = "configName") String configName) throws OHServiceException {
+		LOGGER.info("Retrieve the userSetting By user {} and configName {}:", userName, configName);
+		List<UserSetting> userSettings = userSettingManager.getUserSettingByUserName(userName);
+		if (userSettings == null || userSettings.isEmpty()) {
+		    LOGGER.info("UserSetting not found!");
+		    throw new OHAPIException(new OHExceptionMessage("UserSetting not found."));
+		}
+		UserSetting userSetting = userSettingManager.getUserSettingByUserNameConfigName(userName, configName);
+		if (userSetting == null) {
+		    LOGGER.info("No user settings for the user {}", userName);
+		    return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
+		}
+		return ResponseEntity.ok(userSettingMapper.map2DTO(userSetting));
+	}
 
+	/**
+	 * Deletes a {@link UserSetting}.
+	 * 
+	 * @param id - the id of the userSetting {@link UserSetting} to delete.
+	 * @return {@code true} if the userSetting has been deleted, {@code false} otherwise.
+	 * @throws OHServiceException.
+	 */
+	@DeleteMapping(value = "/users/settings/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<Boolean> deleteUserSetting(@PathVariable(name = "id") int id) throws OHServiceException {
+		Optional<UserSetting> userSetting = userSettingManager.getUserSettingById(id);
+		if (!userSetting.isPresent()) {
+		    LOGGER.info("No user settings with id {}", id);
+		    return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
+		}
+		try {
+			 userSettingManager.deleteUserSetting(userSetting.get());
+	    	} catch (OHServiceException serviceException) {
+	            throw new OHAPIException(new OHExceptionMessage("UserSetting not deleted."));
+	    	}
+		return ResponseEntity.ok(true);
+	}
```

Comment:
here an authenticaed user (e.g. "doctor") can still delete other users settings and should be avoided, hence a check if the user is the same calling the endpoint is needed